### PR TITLE
Prevent analytic expansions from shortcutting Smac Planner feasible paths

### PR DESCRIPTION
**Pull request by SteveMacenski**
Generally helps reduce the shortcutting on approach to goals by setting a maximum cost for traversal for the analytic expansion module to use, multiple turning radius refinement attempts, and an optional cost override when shortcutting is OK on approach when absolutely required.

It also addresses a minor collision detector issue that was resolved in the PR 

Needs: docs update + testing from deployed users

### CODE REVIEW

File: nav2_smac_planner/src/analytic_expansion.cpp [hunk 1]
```diff
@@ -196,6 +245,38 @@ typename AnalyticExpansion<NodeT>::AnalyticExpansionNodes AnalyticExpansion<Node
...
+          break;
+        }
+      }
+
+      // (3) Handle exception: there may be no other option close to goal
+      // if max cost is set too low (optional)
+      if (failure) {
+        if (d < 2.0f * M_PI * goal->motion_table.min_turning_radius &&
+          _search_info.analytic_expansion_max_cost_override)
+        {
```
Comment (harderthan) [inline nav2_smac_planner/src/analytic_expansion.cpp:273]: nitpick.  

```suggestion
          _search_info.analytic_expansion_max_cost_override) {
```

Comment (SteveMacenski) [reply nav2_smac_planner/src/analytic_expansion.cpp:273]: That is incorrect formatting.

**Comment by SteveMacenski** [comment]
- [ ] migration
- [ ] config / readme

**Comment by SteveMacenski** [comment]
Docs: https://github.com/ros-planning/navigation.ros.org/pull/511

**Pull request merged**